app/jobs/notify_job.rb

```diff
@@ -5,18 +5,29 @@ class NotifyJob < ApplicationJob
 
   def perform(notification_id:)
     notification = Notification.kept.includes(:subscription).find(notification_id)
-
     return unless notification.subscription.enabled?
 
     data = ActiveModelSerializers::Adapter.create(NotificationSerializer.new(notification)).to_json
     hmac = Encryptor.hmac(notification.subscription.secret, data)
-
-    response = client.post(notification.subscription.callback_url, data, 'PECS-SIGNATURE': hmac, 'PECS-NOTIFICATION-ID': notification_id)
-
-    notification.delivered_at = DateTime.now if response.success?
+    success = false # assume failure
+
+    begin
+      response = client.post(notification.subscription.callback_url, data, 'PECS-SIGNATURE': hmac, 'PECS-NOTIFICATION-ID': notification_id)
+      if response.success?
+        notification.delivered_at = DateTime.now
+        success = true
+      else
+        puts "Error: non-success status received from #{notification.subscription.callback_url} (#{response.status})"
+      end
+    rescue Faraday::ClientError => e
+      puts "Error notifying #{notification.subscription.callback_url}: #{e.inspect}"
```

Comment:
do we need these puts?
should we not Log in some way instead of just puts it? (maybe sentry?)

Reply:
good point - I'll use the rails.logger instead